### PR TITLE
Changes for Hannibal

### PR DESCRIPTION
**Pull request by amithkanand**

### CODE REVIEW

**Comment by cbaenziger** [comment]
Hi Amit, thanks for getting this moving! Please be sure to move all calls to `bcpc` attributes or bcpc-hadoop utility functions to a BCPC-Hadoop wrapper recipe. Other cookbooks should not depend on `bcpc-hadoop`.

**Comment by cbaenziger** [comment]
A notice from Ritesh: can you please verify the log rotations?

**Comment by cbaenziger** [comment]
Can you please either update the spec tests or remove them?

**Comment by amithkanand** [comment]
@cbaenziger The core `hannibal` cookbook doesn't depend on `bcpc-hadoop`. There is new wrapper cookbook called `bach_hannibal` that was added for deployment on `bach` clusters and `bach_hannibal` does depend on `bcpc-hadoop`.

File: stub-environment/roles/Hannibal-Deploy.json [hunk 1]
```diff
@@ -4,8 +4,12 @@
     },
     "json_class": "Chef::Role",
     "run_list": [
-      "recipe[bcpc-hadoop::hannibal]",
-      "recipe[hannibal::hannibal_deploy]"
+      "recipe[bach_hannibal::default]",
+      "recipe[bach_hannibal::hannibaldb]",
+      "recipe[bach_hannibal::hannibalhb]",
+      "recipe[hannibal::default]",
```
Comment (aespinosa) [inline stub-environment/roles/Hannibal-Deploy.json]: After discussing with Amit:

Let's move all the `hannibal::*` recipes inside the role to `include_recipe 'hannibal::*'` calls inside the `bach_hannibal` wrapper cookbook.

It's better to have the core recipes in `hannibal` to be explicitly depended on to have less surprises.

**Pull request merged**